server/auth.go

```diff
@@ -447,8 +451,80 @@ func (s *Server) processClientOrLeafAuthentication(c *client, opts *Options) boo
 	if s.trustedKeys != nil {
 		if c.opts.JWT == _EMPTY_ {
 			s.mu.Unlock()
-			c.Debugf("Authentication requires a user JWT")
-			return false
+
+			var acc *Account
+			if c.opts.Account == _EMPTY_ {
+				// check default (system account) callout
+				acc = s.SystemAccount()
+				if acc == nil {
+					c.Debugf("Authentication requires a user JWT - Token Request failed - Account Missing")
+					return false
+				}
+			} else {
+				// check account listed
+				acc, err = s.fetchAccount(c.opts.Account)
+				if err != nil {
+					c.Debugf("Authentication requires a user JWT - Token Request Account %q lookup failed: %s", c.opts.Account, err)
+					return false
+				}
+			}
+			acc.mu.RLock()
+			signupName := fmt.Sprintf("Account: %s/%s", acc.Name, acc.nameTag)
+			featOnboard := acc.tags.Contains(featureTagOnboard)
+			acc.mu.RUnlock()
+			if !featOnboard {
+				c.Debugf("Authentication requires a user JWT")
+				return false
+			}
+			// Perform the actual JWT Request
+			msgChan := make(chan []byte)
+			defer close(msgChan)
+			ib := s.newRespInbox()
+			sub, err := s.sysSubscribe(ib, func(_ *subscription, _ *client, acc *Account, subject, reply string, rmsg []byte) {
+				msgChan <- copyBytes(rmsg)
+			})
+			if err != nil {
+				c.Debugf("Authentication requires a user JWT - Setup failure %s", signupName)
+				return false
+			}
+			defer s.sysUnsubscribe(sub)
+			var hdr map[string]string
+			ci := c.getClientInfo(true)
```

Comment:
What will our guidance be on what part of client info will be used to securely identify a new client as being authorized to request access? Would they have something like a JWT set that the callback can inspect?

Reply:
I basically re-use the token value in the jwt context to do just that.
TestJWTSignupFromSysAcc provides nats.Token("shared-secret-bearer") in line 5928 to the connection which is checked by the jwt issuing service in line 5893.

Right now, I'd allow for an empty token, but can be convinced to require one

Reply:
The client info I threw in because it exists. 
(Once a customer wanted to know the ip, a client connected from, in order to issue jwt)